singularity-eos/eos/eos_base.hpp

```diff
@@ -391,8 +414,8 @@ class EosBase {
   void EntropyIsNotEnabled(const char *eosname) const {
     // Construct the error message using char* so it works on device
     char msg[120] = "Entropy is not enabled for the '";
```

Comment:
I know this isn't part of this PR, but we should really consider shrinking this to as small as we can make it.

Reply:
Let's see some rough estimates here... the first half of the message is 32 chars. The longest EOS name is `SpinerEOSDependsRhoSie` which is 22 characters long. Plus the null terminating character. So currently this message could be as small as 55 characters at the moment. But that doesn't include future potential EOS's. 

I'm not comfortable being too too close to the margin there. So maybe we could do something like 81 chars?

Reply:
I forgot the 5 chars for the end of the message. Anyway I updated it here. Message length is now 81 chars, which is still 22 characters longer than it needs to be currently, but I think this is a reasonable margin for error for now.

Reply:
It should be pretty easy to add a check to `StrCat`. I'll do so.

Reply:
@jhp-lanl @dholladay00 please take a look at `StrCat` now. I make the maximum string length a global constant (hidden behind an implementation namespace) and call `PORTABLE_REQUIRE` inside `StrCat`. It'll only yell at you when you compile win "debug mode" but I think that's fine.

Reply:
I approve